code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -62,6 +62,34 @@
 		for(var/mob/O in viewers(user, 3))
 			O.show_message(text("<span class='notice'>\The [src] has been broken by [] with an electromagnetic card!</span>", user), 1, text("You hear a faint electrical spark."), 2)
 			return
+
+/obj/item/weapon/storage/lockbox/emp_act(severity)
+	for(var/obj/O in src)
+		if(istype(O, /obj/item/weapon/gun/energy/stunrevolver))
+			explosion(O.loc,0,1,4) //fuck your shitty stun revolvers
+			qdel(src)
+		var/obj/item/weapon/stock_parts/cell/C
+		for(C in O)
+			C.corrupt()
+			if(prob(80)) //cell is probably kill
+				C.corrupt()
+				C.corrupt()
+			if(prob(13)) //unlucky kaboop
+				C.rigged = 1
+		O.emp_act(severity)
+	if(!src.broken)
+		if(prob(80/severity)) //a decent chance, it is a shitty metal case after all

```

Comment:
I would merge this section with the if prob for the cell breaking section due to the fact that the chance of getting a normal fine weapon after a EMP is very distrusted at the moment due to how bad the manipulation of this mechanic could be.